src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/transport/PerformanceAnalyzerTransportRequestHandler.java

```diff
@@ -43,7 +45,12 @@
 
     @Override
     public void messageReceived(T request, TransportChannel channel, Task task) throws Exception {
-        actualHandler.messageReceived(request, getChannel(request, channel, task) , task);
+        try {
+            actualHandler.messageReceived(request, getChannel(request, channel, task), task);
+        } catch (Exception ex) {
+            LOG.error(ex);
+            StatsCollector.instance().logException(StatExceptionCode.ES_REQUEST_INTERCEPTOR_ERROR);
+        }
```

Comment:
We are consuming an exception that would've otherwise been caught higher up in the stack, does it have any side effects?

Reply:
That is the intent. An exception in the PA code should not lead to exception in ES

Reply:
Thanks for addressing the issue!